xapian-core/api/queryinternal.cc

```diff
@@ -1315,6 +1381,7 @@ QueryValueGE::postlist(QueryOptimiser *qopt, double factor) const
 	RETURN(NULL);
     }
     if (limit < lb) {
+	// should't this be <= ??
```

Comment:
I think you're right.  It means the optimisation won't be applied in the edge case so it's an efficiency problem not a correctness one.  But as James said, that would be better handled as a separate change.